projects/objectnav_baselines/models/object_nav_models.py

```diff
@@ -44,42 +41,76 @@ def __init__(
+        multiple_beliefs=False,
+        beliefs_fusion: Optional[FusionType] = None,
+        auxiliary_uuids: Optional[List[str]] = None,
+        # below are custom params
+        rgb_uuid: Optional[str] = None,
+        depth_uuid: Optional[str] = None,
+        object_type_embedding_dim=8,
+        trainable_masked_hidden_state: bool = False,
+        # perception backbone params,
+        backbone="gnresnet18",
+        resnet_baseplanes=32,
     ):
         """Initializer.
 
         See class documentation for parameter definitions.
         """
-        super().__init__(action_space=action_space, observation_space=observation_space)
+        super().__init__(
+            action_space=action_space,
+            observation_space=observation_space,
+            hidden_size=hidden_size,
+            multiple_beliefs=multiple_beliefs,
+            beliefs_fusion=beliefs_fusion,
+            auxiliary_uuids=auxiliary_uuids,
+        )
 
         self.goal_sensor_uuid = goal_sensor_uuid
         self._n_object_types = self.observation_space.spaces[self.goal_sensor_uuid].n
-        self._hidden_size = hidden_size
         self.object_type_embedding_size = object_type_embedding_dim
 
-        self.visual_encoder = SimpleCNN(
-            observation_space=self.observation_space,
-            output_size=self._hidden_size,
-            rgb_uuid=rgb_uuid,
-            depth_uuid=depth_uuid,
-        )
+        self.backbone = backbone
+        if backbone == "simple_cnn":
+            self.visual_encoder = SimpleCNN(
+                observation_space=observation_space,
+                output_size=hidden_size,
+                rgb_uuid=rgb_uuid,
+                depth_uuid=depth_uuid,
+            )
+        else:  # resnet family
+            self.visual_encoder = resnet.GroupNormResNetEncoder(
+                observation_space=observation_space,
+                output_size=hidden_size,
+                rgb_uuid=rgb_uuid,
+                depth_uuid=depth_uuid,
+                baseplanes=resnet_baseplanes,
+                ngroups=resnet_baseplanes // 2,
+                make_backbone=getattr(resnet, backbone),
+            )
 
-        self.state_encoder = RNNStateEncoder(
-            (0 if self.is_blind else self._hidden_size) + object_type_embedding_dim,
-            self._hidden_size,
-            trainable_masked_hidden_state=trainable_masked_hidden_state,
-            num_layers=num_rnn_layers,
+        self.create_state_encoders(
+            obs_embed_size=self.goal_visual_encoder_output_dims,
+            num_rnn_layers=num_rnn_layers,
             rnn_type=rnn_type,
+            add_prev_actions=add_prev_actions,
+            prev_action_embed_size=action_embed_size,
+            trainable_masked_hidden_state=trainable_masked_hidden_state,
         )
 
-        self.actor = LinearActorHead(self._hidden_size, action_space.n)
-        self.critic = LinearCriticHead(self._hidden_size)
+        self.create_actorcritic_head()
```

Comment:
You might want to make this:
```python
self.actor: LinearActorHead
self.critic: LinearCriticHead
self.create_actorcritic_head()
```
as otherwise code analysis tools might complain about variables being declared outside of `__init__`.

Reply:
Ok. I defined it in the base class now